keep/api/utils/tenant_utils.py

```diff
@@ -15,13 +16,138 @@
 logger = logging.getLogger(__name__)
 
 
+def delete_api_key_internal(
+    session: Session,
+    tenant_id: str,
+    unique_api_key_id: str,
+) -> str:
+    """
+    Deletes API key for the given tenant.
+
+    Args:
+        session (Session): _description_
+        tenant_id (str): _description_
+        unique_api_key_id (str): _description_
+
+    Returns:
+        str: _description_
+    """
+    # Find API key
+    statement = (
+        select(TenantApiKey)
+        .where(TenantApiKey.reference_id == unique_api_key_id)
+        .where(TenantApiKey.tenant_id == tenant_id)
+    )
+
+    api_key = session.exec(statement).first()
+
+    if api_key:
+        # Delete from database
+        api_key.is_deleted = True
+        session.commit()
+
+        # Delete from secret manager
+        context_manager = ContextManager(tenant_id=tenant_id)
+        secret_manager = SecretManagerFactory.get_secret_manager(context_manager)
+
+        secret_manager.delete_secret(
```

Comment:
maybe wrap with try/except and a warning message if fails to delete

Reply:
api: [75a9edb](https://github.com/keephq/keep/pull/752/commits/75a9edbbdc00c4b68c3ef9c5f9f7e2349b58a36c)
ui: toast missing will implement later tonight